fe/fe-core/src/main/java/org/apache/doris/analysis/ArrayLiteral.java

```diff
@@ -134,6 +135,29 @@ protected void toThrift(TExprNode msg) {
         msg.setChildType(((ArrayType) type).getItemType().getPrimitiveType().toThrift());
     }
 
+    @Override
+    public int hashCode() {
+        return Objects.hashCode(children);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == null) {
+            return false;
+        }
+        if (!(o instanceof ArrayLiteral)) {
+            return false;
+        }
+        if (this == o) {
+            return true;
+        }
+        ArrayLiteral that = (ArrayLiteral) o;
+        if (that.children.size() != children.size()) {
+            return false;
+        }
```

Comment:
remove these lines or u need to check children == null first

Reply:
I think Objects.equals make sure the size already! I will just remove this